themes/bootstrap3/js/check_item_statuses.js

```diff
@@ -188,7 +201,7 @@ VuFind.register('itemStatuses', function ItemStatuses() {
   //store the handlers in a "hash" obj
   var checkItemHandlers = {
     ils: makeItemStatusQueue(),
-    overdrive: makeItemStatusQueue({ url: "/Overdrive/getStatus" }),
+    overdrive: makeItemStatusQueue({handlerName: "overdrive"}),
```

Comment:
I'm not sure how much this matters, but I feel that switching to an abstract handlerName instead of passing in a url makes this code potentially less extensible, since the specific knowledge of URLs is now hard-coded into lower-level code instead of passed as a parameter.

Reply:
I changed this because the URL would also have to be added in line 251.  And having the same URL in one place should be easier to manage.

Or is your point that the handlerName passed into `getItemStatusPromise` and you would prefer that the URL is passed by using `getStatusUrl` in both cases the method is called?

Reply:
My feeling is that the ideal situation would be that, if I wanted to add a third status handler in my own local Javascript file, there should be a way to do this without editing check_item_statuses.js. I realize that this is not possible in the existing code, but these edits seem to move us further away from that goal by hard-coding details into code rather than using variables.

One approach could be to add a new array parallel with checkItemHandlers which contains URLs associated with handlers. Then getStatusUrl could look up the URLs in that array. In future, if we wanted to add a public method to register a new handler, it would be pretty easy to do that.

Does that make any sense?

Reply:
Yes, that makes sense. I moved the URLs in an object and also added a public method for adding new handlers.